addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1685,24 +1685,16 @@ options.registry.Carousel = options.Class.extend({
         // the slide overlay) + See "CarouselItem" option.
         this.$controls.addClass('o_we_no_overlay');
 
-        let _slideTimestamp;
-        this.$target.on('slide.bs.carousel.carousel_option', () => {
-            _slideTimestamp = window.performance.now();
-            setTimeout(() => this.trigger_up('hide_overlay'));
-        });
-        this.$target.on('slid.bs.carousel.carousel_option', () => {
-            // slid.bs.carousel is most of the time fired too soon by bootstrap
-            // since it emulates the transitionEnd with a setTimeout. We wait
-            // here an extra 20% of the time before retargeting edition, which
-            // should be enough...
-            const _slideDuration = (window.performance.now() - _slideTimestamp);
-            setTimeout(() => {
-                this.trigger_up('activate_snippet', {
-                    $snippet: this.$target.find('.carousel-item.active'),
-                    ifInactiveOptions: true,
-                });
-                this.$target.trigger('active_slide_targeted');
-            }, 0.2 * _slideDuration);
+        // Handle the sliding manually.
+        this.__onControlClick = _.throttle(this._onControlClick.bind(this), 1000);
+        this.$controls.on("click.carousel_option", this.__onControlClick);
+
+        // Activate the active slide if the content changed.
+        this.$target.on("content_changed.carousel_option", () => {
+            this.trigger_up("activate_snippet", {
```

Comment:
That seems louche. It should at least be with explanations. But is it right to keep calling activate snippets each time we change something?

Reply:
From what I saw, the only place where `content_changed` is triggered is when `removeSlide` is called, are there other situations I am not aware of ? :thinking: 

For the reason why it is needed, it is because `activate_snippet` doesn't seem to work as expected if an element with an editor is removed in the option that triggers it => it always ends up activating the parent instead of the specified element :grimacing: I was planning to discuss it with you to see if it is normal or if it should be fixed :slightly_smiling_face: 
It seems to be a general issue: I tested by modifying other options (like when you remove columns with the layout_column option) and the issue happens too

So here, by doing like this, it allows to activate the active slide after removing one with `removeSlide` => without that, it always activates only the parent snippet (with the `Carousel` option) making the `CarouselItem` option section disappear every time we remove a slide.
So, if we want to remove multiple slides in a row, we have to do like this: `Click on "-" > Click on slide > Click on "-"` instead of `Click on "-" > Click on "-"` which I found really annoying (and it was not the case before so I didn't want to break this behavior)

Note that it is the only way I found to make that work :grimacing:  I guess it was working by chance before because the option  was not properly "mutexed"/"waited" => fixing it therefore revealed this issue :slightly_smiling_face:

Reply:
> From what I saw, the only place where content_changed is triggered is when removeSlide is called, are there other situations I am not aware of ? 🤔

That event is triggered when *any* option is being used. So I am surprised that this does not select the active slide even when an inner column background color is modified for example :open_mouth: 

> For the reason why it is needed, it is because activate_snippet doesn't seem to work as expected if an element with an editor is removed in the option that triggers it 

It's probably an issue indeed :grimacing: Maybe selecting the parent should only be done if nothing is selected anymore? :thinking:

Reply:
In the end, I added the "activate the active slide when a slide is removed" in the destroy of the "CarouselItem" option, with a flag indicating that the destroy happens because the slide was removed. (I couldn't find a better way, but at least this one makes more sense)